ServerCore/Pages/Teams/Details.cshtml.cs

```diff
@@ -31,11 +32,25 @@ public async Task<IActionResult> OnGetAsync(int teamId = -1)
             {
                 return NotFound("Missing team id");
             }
-            else
+
+            // Auth checks (only allow admins OR players who are on the team)
+            int playerTeamId = await GetTeamId();
+            if (!(EventRole == EventRole.admin || (EventRole == EventRole.play && playerTeamId == teamId)))
```

Comment:
This needs to check that the user actually is an admin (this was previously handled by the base class, but that's no longer true due to efficiency issues)